src/Raven.Server/ServerWide/Commands/DeleteDatabaseCommand.cs

```diff
@@ -60,11 +60,17 @@ public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
                             if (record.Sharding.Shards.ContainsKey(ShardNumber.Value) == false)
                                 throw new RachisApplyException($"The requested shard '{ShardNumber}' doesn't exists in '{record.DatabaseName}'");
 
+                            if (record.Sharding.Shards[ShardNumber.Value].ReplicationFactor == 1 && record.Sharding.DoesShardHaveBuckets(ShardNumber.Value))
```

Comment:
`TryGetValue`

Reply:
Also here, the lines above check `ContainsKey`

Reply:
Use `TryGetValue` to replace both checks

Reply:
done